# Phylogenetic-analysis
# Epidemiology and population structure of *Haemophilus influenzae* causing invasive disease

This project was used to understand the population structure of non-typeable *Haemophilus influenzae* (NTHi).

We used R studio and ggplot2 and ggtree packages to observe the phylogenetic distribution of NTHi genomes and its association with metadata, including a clade-related classification based on the presence or abscence of 17 accessory genes (De Chiara, *et al*. Genome sequencing of disease and carriage isolates of NTHi identifies discrete population structure. Proc Natl Acad Sci 2014;111:5439–5444; and Pinto M, *et al*. Insights into the population structure and pan-genome of *H. influenzae*. Infect Genet Evol 2019;67:126–135).
